ide/git/src/org/netbeans/modules/git/ui/history/RevisionNode.java

```diff
@@ -78,31 +94,59 @@ public RevisionNode(RepositoryRevision.Event revision, SearchHistoryPanel master
         initProperties();
     }
 
-    RepositoryRevision getContainer() {
-        return container;
-    }
-
     RepositoryRevision.Event getEvent() {
         return event;
     }
 
+    @Override
+    public String getHtmlDisplayName() {
+        if (isCommitNode()) {
+            return "<b>"+getName()+"</b>";
+        } else {
+            String c = annotationColorForAction(event.getAction());
+            return c != null ? "<font color="+c+">"+getName()+"</font>" : getName();
+        }
+    }
```

Comment:
```suggestion
    @Override
    public String getHtmlDisplayName() {
        if (isCommitNode()) {
            return "<b>" + escape(getName()) + "</b>";
        } else {
            String c = annotationColorForAction(event.getAction());
            return c != null ? "<font color=" + escape(c) + ">" + escape(getName()) + "</font>" : escape(getName());
        }
    }
```

Reply:
RevisionNode is for the diff view tree. It has only commit hashes or file names which is both compatible with html.

Reply:
I would be careful with such predictions:

```
matthias@enterprise:~$ mkdir test
matthias@enterprise:~$ cd test
matthias@enterprise:~/test$ touch '<b>filename with (broken) <br>html styling'
matthias@enterprise:~/test$ ls
'<b>filename with (broken) <br>html styling'
matthias@enterprise:~/test$
```

Yes - this broken HTML, given the age of the HTML parser of swing and the codebases relying on established behavior, I suspect, that it will happily chew this. Given that html and the state of its parsers is a royal mess, I would never trust a html parser to reject anything (regrettably my last hope xhtml died more or less).

Reply:
this column actually doesn't use the JDK html renderer but it doesn't hurt to escape it too. We should consider to upgrade input cleanup to some standard lib like OWASP.

Reply:
> this column actually doesn't use the JDK html renderer but it doesn't hurt to escape it too

Which html renderer is used then?

> We should consider to upgrade input cleanup to some standard lib like OWASP.

No we should not.

You can't "cleanup" input in a sane way or only very limited. What you can do is to limit your input to values that match your target domain. For example if you expect a quantity you can limit inputs to "NUMBER UNIT" pairs or you can expect HTML with only a limited subset of allowed elements or attributes. 

This works for interactive use where the user can validate the final result (if you filter) or adjust the input according to the validation feedback. But we have no interaction here. The data (filename, comment, author name, ....) is provided as is and needs to be rendered as is.

Cleaning up input for unconstraint domains also does not work. Consider: https://xkcd.com/327/. You can argue, that input sanitizing should have caught that, but then I would ask: On which basis do you define the domain "firstname of a person". Somewhere on this planet that first name will be valid and when the person tries to enter your country and some point you will need to deal with that fact.

What is more input sanitizing can't consider the output format. For example for use in HTML the string `Robert'); DROP TABLE Students;--` is perfectly ok, for use in SQL it must be escaped. There is no need for OWASP, because escaping will do the right thing.

Reply:
https://javadoc.io/static/org.owasp.encoder/encoder/1.2.3/org/owasp/encoder/Encode.html

> Which html renderer is used then?

what NB uses for some components:
https://github.com/apache/netbeans/blob/master/platform/openide.awt/src/org/openide/awt/HtmlRenderer.java